File: Swift/LeetCode/LeetCode2033.swift
// https://leetcode.com/problems/minimum-operations-to-make-a-uni-value-grid

class Solution {
    func minOperations(_ grid: [[Int]], _ x: Int) -> Int {
        let flattenGrid = grid.flatMap { $0 }.sorted()
        let remainder = flattenGrid.first! % x
        guard flattenGrid.allSatisfy({ $0 % x == remainder}) else {
            return -1
        }
        let median = flattenGrid[flattenGrid.count / 2]
        return flattenGrid.map { abs($0 - median) / x }.reduce(0) { $0 + $1 }
    }
}

File: python/leetcode/leetcode2033.py
# https://leetcode.com/problems/minimum-operations-to-make-a-uni-value-grid

class Solution:
    def minOperations(self, grid: List[List[int]], x: int) -> int:
        flatGrid = sorted([num for row in grid for num in row])
        remainder = flatGrid[0] % x
        modeList = []
        for element in flatGrid:
            if element % x == remainder:
                continue
            else:
                return -1
        answer = 0
        median = flatGrid[len(flatGrid) // 2]
        for element in flatGrid:
            answer += abs(element - median) // x
        return answer
